repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilterLoader.scala

```diff
@@ -0,0 +1,320 @@
+import org.openrepose.core.filter.SystemModelInterrogator
+import org.openrepose.core.services.classloader.ClassLoaderManagerService
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.services.deploy.ApplicationDeploymentEvent.APPLICATION_COLLECTION_MODIFIED
+import org.openrepose.core.services.deploy.{ApplicationDeploymentEvent, ArtifactManager}
+import org.openrepose.core.services.event.PowerFilterEvent.POWER_FILTER_CONFIGURED
+import org.openrepose.core.services.event.{Event, EventListener, EventService}
+import org.openrepose.core.services.healthcheck.{HealthCheckService, Severity}
+import org.openrepose.core.services.jmx.ConfigurationInformation
+import org.openrepose.core.spring.{CoreSpringProvider, ReposeSpringProperties}
+import org.openrepose.core.systemmodel.config.{Filter => FilterConfig, _}
+import org.openrepose.powerfilter.ReposeFilterLoader._
+import org.openrepose.powerfilter.filtercontext.FilterConfigWrapper
+import org.springframework.beans.factory.NoSuchBeanDefinitionException
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.ApplicationContext
+import org.springframework.context.support.AbstractApplicationContext
+
+import scala.collection.JavaConverters._
+
+@Named("reposeFilterLoader")
+class ReposeFilterLoader @Inject()(@Value(ReposeSpringProperties.NODE.NODE_ID) nodeId: String,
+                                   configurationService: ConfigurationService,
+                                   eventService: EventService,
+                                   healthCheckService: HealthCheckService,
+                                   artifactManager: ArtifactManager,
+                                   applicationContext: ApplicationContext,
+                                   configurationInformation: ConfigurationInformation,
+                                   classLoaderManagerService: ClassLoaderManagerService)
+  extends StrictLogging {
+
+  private val healthCheckServiceProxy = healthCheckService.register
+  private val currentSystemModel = new AtomicReference[Option[SystemModel]](None)
+  private val currentFilterContextRegistrar = new AtomicReference[Option[FilterContextRegistrar]](None)
+  private var currentServletContext = new AtomicReference[Option[ServletContext]](None)
+  private val configurationLock = new Object
+
+  private val systemModelConfigurationListener = new UpdateListener[SystemModel] {
+    private var initialized: Boolean = false
+
+    override def configurationUpdated(configurationObject: SystemModel): Unit = {
+      logger.debug("{} New system model configuration provided", nodeId)
+      val previousSystemModel = currentSystemModel.getAndSet(Option(configurationObject))
+      if (previousSystemModel.isEmpty) {
+        logger.debug("{} -- issuing POWER_FILTER_CONFIGURED event from a configuration update", nodeId)
+        eventService.newEvent(POWER_FILTER_CONFIGURED, System.currentTimeMillis)
```

Comment:
What does this event do?

Reply:
It is used to [start the ArtifactManager's watcher thread](https://github.com/rackerlabs/repose/blob/REP-4381_ReposeFilter/repose-aggregator/core/repose-core/src/main/java/org/openrepose/core/services/deploy/ArtifactManager.java#L91).